cli/npm/managed/cache.rs

```diff
@@ -127,7 +124,26 @@ impl NpmCache {
       .await?;
     match maybe_bytes {
       Some(bytes) => {
-        verify_and_extract_tarball(package, &bytes, dist, &package_folder)
+        let extraction_mode = if should_use_cache {
+          TarballExtractionMode::SiblingTempDir
+        } else {
+          // the user ran with `--reload`, so overwrite the package instead of
+          // deleting it since the package might get corrupted if a user kills
+          // their deno process while it's deleting a package directory
+          TarballExtractionMode::Overwrite
```

Comment:
I don't understand this. If you're worried about deleting leaving it a bad state:

1. rename the real folder to a sibling
2. delete the sibling

ie the reverse of what we do for creation

Reply:
I thought about that, but I don't think that will work if that folder is being used by another process.

Reply:
Or if the folder now has hardlinks (seems like windows doesn't like that)